server/store/sqlstore/group.go

```diff
@@ -0,0 +1,50 @@
+package sqlstore
+
+import (
+	"strings"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+)
+
+var escapeLikeSearchChar = []string{
+	"%",
+	"_",
+}
+
+func sanitizeSearchTerm(term string) string {
+	const escapeChar = "\\"
+
+	term = strings.ReplaceAll(term, escapeChar, "")
+
+	for _, c := range escapeLikeSearchChar {
+		term = strings.ReplaceAll(term, c, escapeChar+c)
+	}
+
+	return term
+}
+
+func (ss SQLStore) SearchGroupsByPrefix(prefix string) ([]*model.Group, error) {
+	sanitizedPrefix := strings.ToLower(sanitizeSearchTerm(prefix))
+	query := ss.replicaBuilder.
+		Select("Id", "DisplayName", "DeleteAt").
+		From("UserGroups").
+		Where(sq.Like{"LOWER(DisplayName)": sanitizedPrefix + "%"}).
```

Comment:
Should we also search the `name` of groups? Keep in mind `name` can be empty

Reply:
My thinking is: if the `DisplayName` is what the user sees, that's what they would expect to specify. Is there a scenario where the user would want to use `Name` instead?

Reply:
The name is what user's see if a group is @ mentioned so they might know it by that. It's usually very similar to the display name though so it's not a big deal

Reply:
done :+1: 

I also attached a video of this in action to the PR description.